src/main/java/org/openstreetmap/atlas/geography/atlas/sqlite/SQLiteItemWriter.java

```diff
@@ -0,0 +1,216 @@
+
+        for (final AtlasItem item : items)
+        {
+            // INSERT feature data
+            final long atlasId = item.getIdentifier();
+            this.itemStatement.setLong(FIRST_COLUMN, atlasId);
+            this.itemStatement.setString(SECOND_COLUMN, "md5sum_here");
+
+            final Optional<String> shardName = atlas.metaData().getShardName();
+            if (shardName.isPresent())
+            {
+                this.itemStatement.setString(THIRD_COLUMN, shardName.get());
+            }
+            else
+            {
+                this.itemStatement.setString(THIRD_COLUMN, null);
+            }
+
+            final byte[] wkb = getWKBOfItem(item);
+            this.itemStatement.setBytes(FOURTH_COLUMN, wkb);
+            this.itemStatement.execute();
+
+            // INSERT tag data for the feature
+            final long tagStoreIndex = this.identifierToTagStoreIndex.get(new Long(atlasId));
+            final int[] keyIndices = this.tagStore.getKeyArrayAtIndex(tagStoreIndex);
+            final int[] valueIndices = this.tagStore.getValueArrayAtIndex(tagStoreIndex);
+
+            if (keyIndices.length != valueIndices.length)
+            {
+                throw new CoreException(
+                        "key and value array length mismatch, something went wrong");
+            }
+
+            for (int index = 0; index < keyIndices.length; index++)
+            {
+                this.tagsStatement.setLong(FIRST_COLUMN, atlasId);
+                this.tagsStatement.setInt(SECOND_COLUMN, keyIndices[index]);
+                this.tagsStatement.setInt(THIRD_COLUMN, valueIndices[index]);
+                this.tagsStatement.execute();
+            }
+
+            // INSERT spatial information if enabled
+            if (this.rtreeStatementOptional.isPresent())
+            {
+                final PreparedStatement unwrappedStatement = this.rtreeStatementOptional.get();
+
+                final Rectangle bounds = item.bounds();
+                final Location lowerLeft = bounds.lowerLeft();
+                final Location upperRight = bounds.upperRight();
+                final double minLon = lowerLeft.getLongitude().asDegrees();
+                final double minLat = lowerLeft.getLatitude().asDegrees();
+                final double maxLon = upperRight.getLongitude().asDegrees();
+                final double maxLat = upperRight.getLatitude().asDegrees();
+
+                // feature id and bounding box in rtree
+                unwrappedStatement.setLong(FIRST_COLUMN, atlasId);
+                unwrappedStatement.setDouble(SECOND_COLUMN, minLon);
+                unwrappedStatement.setDouble(THIRD_COLUMN, maxLon);
+                unwrappedStatement.setDouble(FOURTH_COLUMN, minLat);
+                unwrappedStatement.setDouble(FIFTH_COLUMN, maxLat);
+
+                unwrappedStatement.execute();
+            }
+        }
+
+        this.connection.commit();
+        this.connection.setAutoCommit(true);
```

Comment:
All statements would need to be closed after they have been executed/used.

Reply:
Generally best approach is to use them in a try block like 
`try (PreparedStatement unwrappedStatement = this.rtreeStatementOptional.get())
{
  //do something with the statement
}
`
Because the PreparedStatement is AutoCloseable it will clean up everything for you.